source/core/mu_parser_internal.cc

```diff
@@ -0,0 +1,451 @@
+      if (data.vars.empty())
+        init_muparser();
+
+      for (unsigned int i = 0; i < n_components; ++i)
+        data.vars[i] = p[i];
+      if (n_components != this->n_vars)
+        data.vars[n_components] = time;
+
+      AssertDimension(values.size(), data.parsers.size());
+      try
+        {
+          for (unsigned int component = 0; component < data.parsers.size();
+               ++component)
+            {
+              Assert(dynamic_cast<Parser *>(data.parsers[component].get()),
+                     dealii::StandardExceptions::ExcInternalError());
+              mu::Parser &parser =
+                // We just checked that the pointer is valid so suppress the
+                // clang-tidy check
+                static_cast<Parser &>(*data.parsers[component]); // NOLINT
+              values[component] = parser.Eval();
+            }
+        } // try
+      catch (mu::ParserError &e)
+        {
+          std::cerr << "Message:  <" << e.GetMsg() << ">\n";
+          std::cerr << "Formula:  <" << e.GetExpr() << ">\n";
+          std::cerr << "Token:    <" << e.GetToken() << ">\n";
+          std::cerr << "Position: <" << e.GetPos() << ">\n";
+          std::cerr << "Errc:     <" << e.GetCode() << ">" << std::endl;
+          AssertThrow(false, ExcParseError(e.GetCode(), e.GetMsg()));
+        } // catch
+    }
+
+    // explicit instantiations
+    template class ::internal::FunctionParserCustom::ParserImplementation<1>;
+    template class ::internal::FunctionParserCustom::ParserImplementation<2>;
+    template class ::internal::FunctionParserCustom::ParserImplementation<3>;
+    template class ::internal::FunctionParserCustom::ParserImplementation<4>;
+    template class ::internal::FunctionParserCustom::ParserImplementation<5>;
+    template class ::internal::FunctionParserCustom::ParserImplementation<6>;
+    template class ::internal::FunctionParserCustom::ParserImplementation<7>;
+    template class ::internal::FunctionParserCustom::ParserImplementation<8>;
+    template class ::internal::FunctionParserCustom::ParserImplementation<9>;
+    template class ::internal::FunctionParserCustom::ParserImplementation<10>;
+    template class ::internal::FunctionParserCustom::ParserImplementation<11>;
+    template class ::internal::FunctionParserCustom::ParserImplementation<12>;
+    template class ::internal::FunctionParserCustom::ParserImplementation<13>;
+    template class ::internal::FunctionParserCustom::ParserImplementation<14>;
+    template class ::internal::FunctionParserCustom::ParserImplementation<15>;
+    template class ::internal::FunctionParserCustom::ParserImplementation<16>;
+    template class ::internal::FunctionParserCustom::ParserImplementation<17>;
+    template class ::internal::FunctionParserCustom::ParserImplementation<18>;
+    template class ::internal::FunctionParserCustom::ParserImplementation<19>;
+    template class ::internal::FunctionParserCustom::ParserImplementation<20>;
```

Comment:
I guess when the documentation will be written, the limit of 20 should be indicated somewhere :)

Reply:
Yes, good point!